### PR TITLE
Remove "Catalog" from menu

### LINKED ISSUE #4346: Catalog dashboard panel is empty
## Prerequisites
* [x] Are you running the latest version? - release candidate 1.13.0
* [x] Are you able to consistently reproduce the issue?
* [x] Did you search the issue queue for existing issue? [Search issues](https://github.com/reactioncommerce/reaction/issues)

## Issue Description
The "Catalog" panel in the operator dashboard is empty since `v1.13.0` and should be removed.

## Steps to Reproduce
Please provide starting context, i.e. logged in as a user, configure a particular payment method.
1. Login as an operator
2. Open the "Catalog" dashboard panel

## Possible Solution
This seems to have been caused when we removed revision control as that was the only controller in the Catalog panel.

### Versions
```
Node: 9.11.1
NPM: 5.6.0
Meteor Node: 8.11.2
Meteor NPM: 5.10.0
Reaction CLI: 0.29.0
Reaction: 1.13.0
Reaction branch: release-1.13.0
Docker: 18.03.1-ce
```

### PR DESCRIPTION
**Pull request by willopez**
Resolves #4346  
Impact: minor
Type: chore

## Issue
Removes the "Catalog' menu item from the side panel because there is currently no UI to manage the catalog

## Solution
Remove `register.js` from the catalog plugin

## Breaking changes
N/A

## Testing
1. Boot up reaction
2. Sign in as an administrator and verify the Catalog menu item is NOT displayed.

### CODE REVIEW

**Comment by spencern** [comment]
@willopez does this need a migration to remove the Catalog menu from existing installs?

**Comment by willopez** [comment]
@kieckhafer your are right, I have updated the PR to only remove the menu item registration.

File: imports/plugins/core/catalog/register.js [hunk 1]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (zenweasel) [inline imports/plugins/core/catalog/register.js]: Is there a reason to add this extra blank line?

**Comment by loanlaux** [comment]
Tried to upgrade a shop to 1.14.0 and had to manually edit the registry for `reaction-catalog` in MongoDB. Looks like a migration is necessary here. Should that be in a separate PR? Happy to take care of this one.

**Comment by willopez** [comment]
@loanlaux Thanks for the info, and please go ahead and create a PR.

**Pull request merged**